### PR TITLE
Updates the documentation with BBS examples

### PR DESCRIPTION
**Pull request by lanterno**
This Pull Request aims to assist a little bit in the documentation by addressing some issues. 

Here's a list of all the done changes:
- Replace examples with others that follow specifically the BBS scheme. All the current examples reference generic function names as they exist in the tests. However, the tests work because they depend on `tests/scheme.spec.ts`, so  we.
- Removes `expect` JS function from jest.  The jest library is not easy to set up for typescript (if it's not already installed). Also, it's not required to test the library and function it.
- I reviewed in detail -and tested- the following sections
  - General Usage
  - Composite Proofs
  - Bound Checks using ZKP

I'm working on testing and improving the other sections, but I didn't want to make a single big PR, to make it easy to review.

### CODE REVIEW

File: README.md
Comment (ineiti) [inline README.md:80]: Please add a short disclaimer here about BBS/BBS+, similar to what is written in the [IRTF draft](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html), but shorter, and mentioning BBS and BBS+:

> The name BBS is derived from the authors of the original academic work by Dan Boneh, Xavier Boyen, and Hovav Shacham [[BBS04](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#BBS04)], where the scheme was first described as part of a group signatures protocol. Soon after, the scheme was described by Camenisch and Lysyanskaya as a stand-alone signatures scheme in [[CL04](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#CL04)], for anonymous credentials applications. Later, Au, Susilo an Mu presented the first, provably secure version of BBS Signatures in [[ASM06](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#ASM06)]. Following, works by Camenisch, Drijvers and Lehmann [[CDL16](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#CDL16)] and by Barki, Brunet, Desmoulins and Traore [[BBDT16](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#BBDT16)], proved the security of the scheme in settings where more efficient computations are possible, thereby improving performance. Finally, in 2023, Tessaro and Zhu, presented in [[TZ23](https://identity.foundation/bbs-signature/draft-irtf-cfrg-bbs-signatures.html#TZ23)] further performance improvements, shrinking the BBS signature. This document is mainly based on that work.

File: README.md
Comment (ineiti) [inline README.md:85]: Either

A BBS signature also allows

or

BBS signatures also allow

File: README.md
Comment (ineiti) [inline README.md:184]: Same as above - but it seems that most other usages are `BBS signatures`, so I'd stick with the plural throughout the document, except where it doesn't make sense...

File: README.md
Comment (ineiti) [inline README.md:254]: Either keep the `expect`, or do something like:

```
if (!result.verified){
  throw new Error("Signature verification failed");
}
```

File: README.md
Comment (ineiti) [inline README.md:261]: plural

File: README.md
Comment (ineiti) [inline README.md:357]: As above, keep the `expect`, or add a `throw`.

File: README.md
Comment (ineiti) [inline README.md:527]: `expect` or `throw`

File: README.md
Comment (ineiti) [inline README.md:829]: `expect` or `throw`

File: README.md
Comment (ineiti) [inline README.md:858]: `expect` or `throw`

**Review by ineiti — COMMENTED** [review]
Great - I put some small comments in there - mainly plural/singular for `BBS signatures` and verification of the signature should either `expect` or `throw` an error.

File: README.md
Comment (lovesh) [reply README.md:184]: Either "verify a BBS signature" or "verify BBS signatures".

File: README.md
Comment (lovesh) [reply README.md:254]: Even I prefer keeping an "error throwing" mechanism over a comment even for illustration purposes. Wdyt about `console.assert` over `expect`? The former is available  in browsers and node.js

File: README.md
Comment (lovesh) [inline README.md:786]: This is a place where BBS and BBS+ differ. With BBS, there is no `blinding` (the commitment is computationally hiding, though it can be made perfectly hiding by adding a dummy attribute). Can you please mention it here. And see comment below on `generateRequest`

File: README.md
Comment (lovesh) [inline README.md:792]: `BBSBlindSignature.generateRequest` returns a single item, i.e. `request` but `BBSPlusBlindSignatureG1.generateRequest` returns as above.

File: README.md
Comment (lovesh) [inline README.md:812]: There are helpers for creating `Witness` object easy for blind signatures. See `getBBSWitnessForBlindSigRequest`, `getBBSPlusWitnessForBlindSigRequest`, `getPSWitnessesForBlindSigRequest`. See usage [here](https://github.com/docknetwork/crypto-wasm-ts/blob/master/tests/composite-proofs/social-kyc.spec.ts#L112).

File: README.md
Comment (lovesh) [inline README.md:840]: Unblinding does not apply to BBS. See [this](https://github.com/docknetwork/crypto-wasm-ts/blob/master/tests/composite-proofs/social-kyc.spec.ts#L134).

**Review by lovesh — CHANGES_REQUESTED** [review]
Mostly related to blind signartures

**Comment by lovesh** [comment]
Thanks @lanterno for these fixes and @ineiti for your comments. Have suggested a few changes.

File: README.md
Comment (ineiti) [reply README.md:254]: I didn't know of `console.assert` - I like it. Yes, let's replace the `throw`/`expect` with `console.assert`.

**Comment by lanterno** [comment]
@ineiti @lovesh  Hello, I've added many improvements on the language as mentioned above, and working on the other points, but I wanted to confirm one of the linguistic points I've changed in the last commit. 

That is:
many times "BBS signatures" is used but means "BBS signature scheme" while in other cases it really means "BBS signatures". So, I just to only use BBS when we mean the overarching scheme and use BBS signature/s otherwise.

So I started the documentation with this section 

> The BBS signature Scheme (henceforth referred to as just BBS)...

Then after use BBS whenever the scheme is meant.

File: README.md
Comment (lanterno) [reply README.md]: Sounds good! I used `console.assert(..);`

**Comment by lovesh** [comment]
@lanterno 

> Then after use BBS whenever the scheme is meant. What do you think?

Sounds good.

**Comment by lanterno** [comment]
@lovesh I updated the blinding part to use BBS+ and added a disclaimer at the top about the difference between BBS and BBS+ when it comes to blinding.  
This isn't perfect since we wanted to use only one crypto scheme through out all the examples, but it might be good to show the good side of BBS+ as well.

File: README.md
Comment (lanterno) [reply README.md]: I added a simple disclaimer at the top. Do you think there's something more to add there?

**Comment by lovesh** [comment]
@lanterno Thank you!.

**Comment by lanterno** [comment]
@lovesh Can you approve the workflow? (and maybe merge)

P.S. @ineiti can't approve PRs, he can only comment. (In case, you're waiting for this approval)

**Comment by lovesh** [comment]
Thanks @lanterno .

**Pull request merged**
